src/plugins/auto-moderation-plugin.ts

```diff
@@ -0,0 +1,161 @@
+import fs from 'node:fs'
+
+import type { PluginContext, PluginInterface } from '../common/plugins.js'
+
+import { EventType, InstanceType, PunishmentType, Severity, ChannelType } from 'src/common/application-event.js'
+import { antiSpamString } from 'src/util/shared-util.js'
+
+enum HeatType {
+  SPAM = 'spam',
+  PROFANITY = 'profanity'
+}
+
+type HeatScore = number
+
+type HeatsTable = Record<HeatType, HeatScore>
+type WarnsTable = Record<HeatType, boolean>
+type InfractionsTable = Record<HeatType, number>
+
+type HeatUser = string
+
+interface InfractionCase {
+  userName: string
+  userUuid: string | undefined
+  userDiscordId: number | undefined
+  infractions: InfractionsTable
+  lastInfraction: number
+}
+
+const heats: Record<HeatUser, HeatsTable> = {}
+const warns: Record<HeatUser, WarnsTable> = {}
+let infractions: InfractionCase[] = []
+
+setInterval(() => {
+  reduceHeats()
+}, 1000)
+
+function saveConfig(path: string): void {
+  const dataRaw = JSON.stringify(infractions, undefined, 4)
+  fs.writeFileSync(path, dataRaw, { encoding: 'utf8' })
+}
+
+function reduceInfractions() {
+  for (const infractionCase of infractions) {
+    if (infractionCase.lastInfraction <= Date.now() - 7 * 24 * 60 * 60 * 1000) {
+      infractions = infractions.filter((infraction) => infraction.userName !== infractionCase.userName)
+    }
+  }
```

Comment:
What is this witchcraft. You can't modify a list while you are looping through it. This will result in undefined behavior.
I assume you want to filter out old infractions.

This suggested code hasn't been tested. 
```suggestion
    infractions = infractions.filter((infractionCase) => infractionCase.lastInfraction > Date.now() - 7 * 24 * 60 * 60 * 1000)
```